pkg/tinydtls/patches/0002-Rename-condition-WITH_RIOT_GNRC-to-WITH_RIOT.patch

```diff
@@ -0,0 +1,60 @@
+From 3e7e414279500bc3061c771e14cb57565ccaf327 Mon Sep 17 00:00:00 2001
+From: HendrikVE <hendrik1995@msn.com>
+Date: Sat, 5 Mar 2022 23:35:09 +0100
+Subject: [PATCH] Rename condition WITH_RIOT_GNRC to WITH_RIOT
+
+Using tinydtls on RIOT OS is not limited to GNRC as network stack.
+It is also working with e.g. lwIP, see: https://github.com/RIOT-OS/RIOT/pull/17552
+Therefore the name WITH_RIOT_GNRC is misleading.
+---
+ session.c | 2 +-
+ session.h | 6 +++---
+ 2 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/session.c b/session.c
+index 300224e..8e9f544 100644
+--- a/session.c
++++ b/session.c
+@@ -31,7 +31,7 @@
+    && (A)->port == (B)->port					\
+    && uip_ipaddr_cmp(&((A)->addr),&((B)->addr))			\
+    && (A)->ifindex == (B)->ifindex)
+-#elif defined(WITH_RIOT_GNRC)
++#elif defined(WITH_RIOT)
```

Comment:
```suggestion
#elif defined(RIOT_VERSION)
````

should work as well and doesn't require a custom define (this is used by other pkgs too)

Reply:
Ah wait so when RIOT is used with the `posix_sockets` module this path should *not*  be taken - then how about `WITH_RIOT_SOCK`

Reply:
I renamed `WITH_RIOT_GNRC` to `WITH_RIOT_SOCK` as suggested.